sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -102,10 +102,10 @@ __esimd_abs_common_internal(simd<TArg, SZ> src0) {
 }
 
 template <typename TRes, typename TArg>
-ESIMD_NODEBUG ESIMD_INLINE
-    std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
-                         detail::is_esimd_scalar<TArg>::value,
-                     TRes>
+ESIMD_NODEBUG
+    ESIMD_INLINE std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
```

Comment:
minor comment. If you change this place, please use this macro that is used in most of other places already:
#define __ESIMD_API ESIMD_NODEBUG ESIMD_INLINE

```suggestion
__ESIMD_API std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
```